Mathlib/Analysis/Calculus/IteratedDeriv/Lemmas.lean

```diff
@@ -123,3 +125,43 @@ lemma iteratedDeriv_comp_neg (n : ℕ) (f : 𝕜 → F) (a : 𝕜) :
     rw [iteratedDeriv_succ, iteratedDeriv_succ, ih', pow_succ', neg_mul, one_mul,
       deriv_comp_neg (f := fun x ↦ (-1 : 𝕜) ^ n • iteratedDeriv n f x), deriv_const_smul',
       neg_smul]
+
+lemma iteratedDeriv_eq_on_open (n : ℕ) {f g : 𝕜 → F} {s : Set 𝕜} (hs : IsOpen s) (x : s)
+    (hfg : Set.EqOn f g s) : iteratedDeriv n f x = iteratedDeriv n g x := by
+  induction' n with n IH generalizing f g
+  · simpa only [iteratedDeriv_zero] using hfg x.2
+  · simp only [iteratedDeriv_succ']
+    exact IH fun y hy ↦ Filter.EventuallyEq.deriv_eq <|
+      Filter.eventuallyEq_iff_exists_mem.mpr ⟨s, IsOpen.mem_nhds hs hy, hfg⟩
```

Comment:
Declaring `x` as an element of `s` coerced to a subtype seems nonoptimal, we should probably go for ` {x : 𝕜} (hx : x ∈ s)`. 

I also feel this should be formulated using "EventuallyEq"; maybe we should add something like
```lean
lemma Filter.EventuallyEq.iteratedDeriv_eq (n : ℕ) {f g : 𝕜 → F} {x : 𝕜}
    (hfg : f =ᶠ[nhds x] g) : iteratedDeriv n f x = iteratedDeriv n g x := by
  simp only [← iteratedDerivWithin_univ, iteratedDerivWithin_eq_iteratedFDerivWithin]
  rw [(hfg.filter_mono nhdsWithin_le_nhds).iteratedFDerivWithin_eq hfg.eq_of_nhds n]
```
instead? Possibly going via the `iteratedDerivWithin` version (which would need `f x = g x` as a hypothesis).